frontend/GW_UI/GW_UI/UserControls/MenuButton.xaml.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Input;
+
+namespace GW_UI.UserControls
+{
+    public partial class MenuButton : UserControl
+    {
+        // Регистрация RoutedEvent для Click
+        public static readonly RoutedEvent ClickEvent = EventManager.RegisterRoutedEvent(
+            "Click", // Имя события
+            RoutingStrategy.Bubble, // Стратегия маршрутизации
+            typeof(RoutedEventHandler), // Тип делегата
+            typeof(MenuButton)); // Владелец события
+
+        // CLR-событие обертка для RoutedEvent
+        public event RoutedEventHandler Click
+        {
+            add { AddHandler(ClickEvent, value); }
+            remove { RemoveHandler(ClickEvent, value); }
+        }
+
+        public MenuButton()
+        {
+            InitializeComponent();
+            this.MouseUp += OnMouseUp;
```

Comment:
You are subscribing to the `MouseUp` event, but I don't see an unsubscription in the code.
If you don’t unsubscribe from events, it can lead to memory leaks :(